lib/core/treenode.js

```diff
@@ -201,6 +204,24 @@ class TreeNode {
         rejectFn(err);
       });
   }
+
+  logCommandResult(result) {
+    let reporter = this.context.reporter;
```

Comment:
@beatfactor I moved this logic to `treenode`. This part is here to get the reporter object from the node. Should I try to add reporter object to a node in a more consistent manner?